lib/elixir/lib/base.ex

```diff
@@ -619,11 +722,137 @@ defmodule Base do
     string |> remove_ignored(opts[:ignore]) |> decode64url!(pad?)
   end
 
+  @doc """
+  Validates a base 64 encoded string with URL and filename safe alphabet.
+
+  ## Options
+
+  Same as `url_decode64/2`.
+
+  ## Examples
+
+      iex> Base.valid_url_decode64?("_3_-_A==")
+      true
+
+      iex> Base.valid_url_decode64?("_3_-_A==\\n", ignore: :whitespace)
+      true
+
+      iex> Base.valid_url_decode64?("_3_-_A", padding: false)
+      true
+
+  """
+  @doc since: "1.19.0"
+  @spec valid_url_decode64?(binary, ignore: :whitespace, padding: boolean) :: boolean
+  def valid_url_decode64?(string, opts \\ []) when is_binary(string) do
+    pad? = Keyword.get(opts, :padding, true)
+    string |> remove_ignored(opts[:ignore]) |> validate64url!(pad?)
+    true
+  rescue
+    ArgumentError -> false
+  end
+
   for {base, alphabet} <- [base: b64_alphabet, url: b64url_alphabet] do
-    name = :"decode64#{base}!"
+    decode_name = :"decode64#{base}!"
+    validate_name = :"validate64#{base}!"
     {min, decoded} = alphabet |> Enum.with_index() |> to_decode_list.()
 
-    defp unquote(name)(char) do
+    defp unquote(validate_name)(char) when char in unquote(alphabet), do: char
```

Comment:
For validating each bit, we should use the same `decode/1` function instead of introducing a new validate function. The decode function is a subtraction plus a tuple operation, which is likely faster than checking for `char in unquote(alphabet)`.

Reply:
This should apply to all of them.

Reply:
Done!

Reply:
No, this is not done :D My suggestion is to remove `defp unquote(validate_name)(char)` and use the decoding function instead, which is likely faster?